distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/ActionsBuilder.java

```diff
@@ -0,0 +1,52 @@
+package com.gruppo1.distributednetworkmanager;
+
+import com.dezen.riccardo.networkmanager.Resource;
+import com.dezen.riccardo.smshandler.Peer;
+
+/**
+ *
+ * @param <T> the real class that describes an Action of you custom network
+ * @param <P> a real type of peer that can be contacted
+ * @param <R> a resource type
+ * @param <N> a Node of the network
+ */
+public interface ActionsBuilder<T, P extends Peer, R extends Resource, N extends Node> {
+
+    T buildPing(P peer);
+
+
+    /**
+     * Hypothesis: this type of responses (ANSW) is built starting from the request (mainly to retrieve the action ID/number)
+     *
+     * QUESTION: is it good?
```

Comment:
In general, yes, it's very unlikely you'll need to build a Response without the Request. Outside of testing at least. You read the Peer for an incoming Action as the Peer for the SMS, which is exactly who you wanna answer to.

Reply:
As I thought👍